File: leetcode/cpp/SingleNumber.cpp
class Solution {
public:
    int singleNumber(int A[], int n) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        int x = 0;
        for (int i = 0; i < n; ++ i) {
            x ^= A[i];
        }
        
        return x;
    }
};

/*


Single Number
Total Accepted: 4471 Total Submissions: 9907

Given an array of integers, every element appears twice except for one. Find that single one.

Note:
Your algorithm should have a linear runtime complexity. Could you implement it without using extra memory? 
*/

File: leetcode_java/SingleNumber.java
public class Solution {
    public int singleNumber(int[] nums) {
        int val = 0;
        for (int i = 0, len = nums.length; i < len; ++ i) {
            val ^= nums[i];
        }
        
        return val;
    }
}

/**
Single Number
Difficulty: Medium

Given an array of integers, every element appears twice except for one. 
Find that single one.

Note:
Your algorithm should have a linear runtime complexity. 
Could you implement it without using extra memory? 
 */
